File: 0452-Minimum Number of Arrows to Burst Balloons/0452-Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<pair<int, int>>& points) {
        int n = points.size();
        if (n == 0) {
            return 0;
        }
        sort(points.begin(), points.end());
        int count = 1;
        int prev = points[0].second;
        for (int i = 1; i < n; ++i) {
            if (points[i].first > prev) {
                ++count;
                prev = points[i].second;
            }
            else {
                prev = min(prev, points[i].second);
            }
        }
        
        return count;
    }
};

File: 0401-0500/0452-Minimum Number of Arrows to Burst Balloons/0452-Minimum Number of Arrows to Burst Balloons.py
import math
class Solution:
    def findMinArrowShots(self, points: 'List[List[int]]') -> 'int':
        points.sort()
        count = 0
        prev = -math.inf
        for point in points:
            if point[0] > prev:
                count += 1
                prev = point[1]
            else:
                prev = min(prev, point[1])
        return count
